### PR TITLE
Automated cherry pick of #65382: Add Reconcile label to kube namespaces

### PR DESCRIPTION
**Pull request by ostromart**
Cherry pick of #65382 on release-1.11.

#65382: Add Reconcile label to kube namespaces

### CODE REVIEW

File: cluster/addons/istio/auth/istio-auth.yaml [hunk 1]
```diff
@@ -3,13 +3,15 @@ kind: Namespace
 metadata:
   name: kube-public
   labels:
+    addonmanager.kubernetes.io/mode: Reconcile
     istio-injection: disabled
 ---
 apiVersion: v1
 kind: Namespace
 metadata:
   name: kube-system
```
Comment (liggitt) [inline cluster/addons/istio/auth/istio-auth.yaml]: This seems odd.  Does this add-on own this namespace? If not, how is it able to apply labels without removing labels set by other components or the admin?

**Comment by jberkus** [comment]
given that the PR against master is not accepted and has unresolved review comments, this PR is not accepted for 1.11.0.  Possibly it will go into 1.11.1 if all issues are resolved and it's approved by the SIG.

**Comment by foxish** [comment]
/close

Making it easier to track for me.
@ostromart, please reopen the cherrypick once the master change lands.

**Pull request closed without merging**